src/components/decimal/decimal.component.js

```diff
@@ -227,17 +227,24 @@ const Decimal = ({
     prevControlledRef.current = isControlled;
   }, [isControlled]);
 
+  const prevValue = usePrevious(value);
+  const componentShouldUpdate = stateValue === value;
+
   useEffect(() => {
     const unformattedValue = toStandardDecimal(stateValue);
 
     if (isControlled) {
       const valueProp = getSafeValueProp(value);
       if (unformattedValue !== valueProp) {
-        setStateValue(formatValue(value));
+        setStateValue(formatValue(valueProp));
+
+        if (valueProp === "" && prevValue === "") {
+          setStateValue(formatValue(emptyValue));
+        }
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [value]);
+  }, [value, componentShouldUpdate]);
```

Comment:
I think we should remove the disable lint and add all the dependencies here, or at least replace `componentShouldUpdate` with `stateValue`

```
  }, [value, stateValue, emptyValue, formatValue, getSafeValueProp, isControlled, prevValue, toStandardDecimal]);
```

Reply:
It will likely lead to the test breaking that we discussed on slack, lets catch up tomorrow and we can look at updating it as I'm not convinced it's testing controlled behaviour like we want